src/sql/workbench/parts/notebook/browser/models/notebookInput.ts

```diff
@@ -385,6 +392,9 @@ export class NotebookInput extends EditorInput {
 	}
 
 	public dispose(): void {
+		if (this._model) {
+			this._model.editorModel.textEditorModel.onBeforeDetached();
```

Comment:
Can you get a null ref here? Maybe comment why this works here but not in the above scenarios/